src/Bicep.Core.IntegrationTests/ExamplesTests.cs

```diff
@@ -39,10 +40,16 @@ private void RunExampleTest(EmbeddedFile embeddedBicep, IFeatureProvider feature
             var bicepFile = baselineFolder.EntryFile;
             var jsonFile = baselineFolder.GetFileOrEnsureCheckedIn(Path.ChangeExtension(embeddedBicep.FileName, jsonFileExtension));
 
-            var dispatcher = new ModuleDispatcher(BicepTestConstants.RegistryProvider);
-            var configuration = BicepTestConstants.BuiltInConfigurationWithAllAnalyzersDisabled;
-            var sourceFileGrouping = SourceFileGroupingBuilder.Build(BicepTestConstants.FileResolver, dispatcher, new Workspace(), PathHelper.FilePathToFileUrl(bicepFile.OutputFilePath), configuration);
-            var compilation = new Compilation(features, new DefaultNamespaceProvider(BicepTestConstants.AzResourceTypeLoader, features), sourceFileGrouping, configuration,BicepTestConstants.ApiVersionProvider, new LinterAnalyzer(configuration));
+            var configManager = IConfigurationManager.ForConfiguration(BicepTestConstants.BuiltInConfigurationWithAllAnalyzersDisabled);
+            var dispatcher = new ModuleDispatcher(BicepTestConstants.RegistryProvider, configManager);
+            var sourceFileGrouping = SourceFileGroupingBuilder.Build(BicepTestConstants.FileResolver, dispatcher, new Workspace(), PathHelper.FilePathToFileUrl(bicepFile.OutputFilePath));
+            var compilation = new Compilation(
+                IFeatureProviderManager.ForFeatureProvider(features),
+                new DefaultNamespaceProvider(BicepTestConstants.AzResourceTypeLoader),
+                sourceFileGrouping,
+                configManager,
+                BicepTestConstants.ApiVersionProvider,
+                new LinterAnalyzer());
```

Comment:
It feels like `LinterAnalyzer` has now become a singleton since you've removed the dependency on configuration, and it appears thread-safe as there's no mutable state. Is it worth following the same pattern as we do for `ApiVersionProvider` and using DI in LangServer/CLI. Looking at references to the constructor, it feels like this should simplify things a fair bit.

Feel free to create an issue if you don't want to address this right now.